pulsar-broker/src/main/java/org/apache/pulsar/broker/zookeeper/aspectj/ClientCnxnAspect.java

```diff
@@ -0,0 +1,181 @@
+            return EventType.write;
+        } else if (response instanceof DeleteRequest) {
+            return EventType.write;
+        } else if (response instanceof SetDataRequest) {
+            return EventType.write;
+        } else if (response instanceof SetACLRequest) {
+            return EventType.write;
+        } else if (response instanceof SetMaxChildrenRequest) {
+            return EventType.write;
+        } else if (response instanceof SetSASLRequest) {
+            return EventType.write;
+        } else if (response instanceof SetWatches) {
+            return EventType.write;
+        } else if (response instanceof SyncRequest) {
+            return EventType.write;
+        } else if (response instanceof ExistsRequest) {
+            return EventType.read;
+        } else if (response instanceof GetDataRequest) {
+            return EventType.read;
+        } else if (response instanceof GetMaxChildrenRequest) {
+            return EventType.read;
+        } else if (response instanceof GetACLRequest) {
+            return EventType.read;
+        } else if (response instanceof GetChildrenRequest) {
+            return EventType.read;
+        } else if (response instanceof GetChildren2Request) {
+            return EventType.read;
+        } else if (response instanceof GetSASLRequest) {
+            return EventType.read;
+        } else {
+            return EventType.other;
+        }
+    }
+
+    private long getStartTime(Object packet) {
+        try {
+            if (packet.getClass().getName().equals("org.apache.zookeeper.ClientCnxn$Packet")) {
+                Field ctxField = Class.forName("org.apache.zookeeper.ClientCnxn$Packet").getDeclaredField("ctx");
+                ctxField.setAccessible(true);
+                Object zooworker = ctxField.get(packet);
+                if (zooworker.getClass().getName().equals("org.apache.bookkeeper.zookeeper.ZooWorker")) {
+                    Field timeField = Class.forName("org.apache.bookkeeper.zookeeper.ZooWorker")
+                            .getDeclaredField("startTimeMs");
+                    timeField.setAccessible(true);
+                    long startTime = (long) timeField.get(zooworker);
+                    return startTime;
+                }
+            }
+        } catch (Exception e) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Failed to get event-start-time from zk-response", e);
+            }
+        }
+        return -1;
+    }
+
+    private Record getEventType(Object packet) {
+        try {
+            if (packet.getClass().getName().equals("org.apache.zookeeper.ClientCnxn$Packet")) {
+                Field field = Class.forName("org.apache.zookeeper.ClientCnxn$Packet").getDeclaredField("request");
+                field.setAccessible(true);
+                Record response = (Record) field.get(packet);
+                return response;
+            }
+        } catch (Exception e) {
```

Comment:
do you think we should do `warn` logging here?
actually, as we are using reflection and if in future field-name or anything changed which throws exception then it can entirely fill out logs so, I made it as debug-log.